onnxruntime/core/graph/function.cc

```diff
@@ -247,11 +248,23 @@ FunctionImpl::FunctionImpl(const onnxruntime::Graph& graph,
                            const ONNX_NAMESPACE::FunctionProto& onnx_func_proto,
                            const logging::Logger& logger)
     : parent_graph_(&graph),
-      body_(onnx_func_proto.name(), false, onnxruntime::ModelMetaData(),
-            graph.ModelPath().ToPathString(),
-            IOnnxRuntimeOpSchemaRegistryList(),
-            graph.DomainToVersionMap(), {}, logger),
       onnx_func_proto_(onnx_func_proto) {
+  std::unordered_map<std::string, int> function_opset_imports{graph.DomainToVersionMap()};
+  for (const auto& opset_import : onnx_func_proto_.opset_import()) {
+    auto it = function_opset_imports.find(opset_import.domain());
+    if (it != function_opset_imports.end()) {
+      ORT_ENFORCE(it->second == opset_import.version(),
+                  "ONNX model does not support multiple opsets for a domain.");
+    } else {
+      function_opset_imports[opset_import.domain()] = static_cast<int>(opset_import.version());
```

Comment:
Suggest to simply insert or emplace into the map and ORT_ENFORCE to make sure it inserted.